fea-rs/src/compile/feature_writer.rs

```diff
@@ -93,6 +99,58 @@ impl<'a> FeatureBuilder<'a> {
             .insert(class_name.into(), MarkClass { members });
     }
 
+    /// Define a mark class with a variable anchor.
+    ///
+    /// Roughly equivalent to `markClass glyph|class <anchor x y> @name{};`
+    /// [markClass](https://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#4.f)
+    /// with a variable anchor record.
+    pub fn add_mark_class_for_glyph(
```

Comment:
The function above is an attempt to provide this same functionality. And I agree there's room for improvement.

I find the relationship between FEA text and the actual GPOS/GSUB/GDEF tables eternally confusing. With that in mind, I'm just going to think through the differences out loud:

- My version expects to be called once, with everything computed; your version expects to be called once per glyph, building up the class. both seem... fine.
- My version requires you to construct an anchor table manually, and also allows for null anchors (since these are allowed in the final table, although I don't know what it means) and yours hides that; hiding it seems good
- My version only supports glyph classes, and your version only supports individual glyphs. I took a look and under the covers I'm always just enumerating the glyph classes and cloning the anchors anyway, so I prefer your version here.
- I agree that, overall, it would be nice to hide the details of anchor construction, but I would prefer to do this with a builder; in particular this will let us support [AnchorFormat2](https://learn.microsoft.com/en-us/typography/opentype/spec/gpos#anchor-table-format-2-design-units-plus-contour-point).